### PR TITLE
[TT-4571] Add name field to auth config

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Add `name` field to auth config to match security scheme name in OAS.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3888/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1838138826)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -597,6 +597,7 @@ type UptimeTests struct {
 }
 
 type AuthConfig struct {
+	Name              string          `mapstructure:"name" bson:"name" json:"name"`
```
Comment (jeffy-mathew) [inline apidef/api_definitions.go:600]: should we name it `AuthConfigName` or just `Name`? it's named `AuthConfigName` in the contract

**Review by jeffy-mathew — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3888/merge
Commit: 3f273f72a37d5c5360433045db9e74b6017cdc49 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1847339728)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/heads/master
Commit: 6e2326820dedd59bf1e773025979d676c9e64456 Add name field to auth config (#3888)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1851573609)

**Pull request merged**